plugins-compat-tester-model/src/main/java/org/jenkins/tools/test/model/hook/PluginCompatTesterHooks.java

```diff
@@ -32,14 +39,37 @@ public PluginCompatTesterHooks() {
         this(new ArrayList<>());
     }
     public PluginCompatTesterHooks(List<String> extraPrefixes) {
-        if(extraPrefixes != null) {
-            hookPrefixes.addAll(extraPrefixes);
-        }
+        setupPrefixes(extraPrefixes);
+        setupHooksByType();
+    }
+
+    public PluginCompatTesterHooks(List<String> extraPrefixes, List<File> externalJars) throws MalformedURLException {
+        setupPrefixes(extraPrefixes);
+        setupExternalClassLoaders(externalJars);
+        setupHooksByType();
+    }
+
+    private void setupHooksByType() {
         for(String stage : Arrays.asList("checkout", "execution", "compilation")) {
             hooksByType.put(stage, findHooks(stage));
         }
     }
+    private void setupPrefixes(List<String> extraPrefixes) {
+        if(extraPrefixes != null) {
+            hookPrefixes.addAll(extraPrefixes);
+        }
+    }
 
+    private void setupExternalClassLoaders(List<File> externalJars) throws MalformedURLException {
+        if (externalJars == null) {
+            return;
+        }
+        for (File jar : externalJars) {
+            if (jar.exists() && jar.isFile()) {
```

Comment:
it would be nice if the tool bailed out if I passed a incorrect filename rather than silently succeeding (but not doing what I expected of it).  could be validate upstream in the options with a FileValidator or here.

Reply:
Validation included on https://github.com/jenkinsci/plugin-compat-tester/pull/300/commits/f11146d232445e7bb202d8d27f17bf93370a0940